### ISSUE TITLE
Bug when deletion overlaps with stop/last AA

### ISSUE THREAD

**Issue by visze** [charite/jannovar#361]
https://github.com/charite/jannovar/blob/44333688c7c319623761478cd5d7d210fbc4d8dc/jannovar-core/src/main/java/de/charite/compbio/jannovar/annotation/builders/DeletionAnnotationBuilder.java#L252

```java
aaChange.getPos()
```
is ZERO based! So you cannot compare the last zero-based position with the length of the transcript wit `==`. I guess it should be:
```java
 if (aaChange.getPos() +1 >= varAASeq.length()) { 
```
Or use maybe use `lastPos()` to check if the deletion ends after the transkript (you checked already that it looses the stop). I guess this is the case you want to handle here...

This change will affect this lin here  in the tests:
https://github.com/charite/jannovar/blob/44333688c7c319623761478cd5d7d210fbc4d8dc/jannovar-core/src/test/java/de/charite/compbio/jannovar/annotation/builders/DeletionAnnotationBuilderTest.java#L1286

**Comment by holtgrewe** [comment]
Good catch!


Am 08.09.2017 2:25 nachm. schrieb Max <notifications@github.com>:

Assigned #361<https://github.com/charite/jannovar/issues/361> to @holtgrewe<https://github.com/holtgrewe>.

**Comment by visze** [comment]
Oh sorry. I mean the line 246! where we have the `==`

https://github.com/charite/jannovar/blob/44333688c7c319623761478cd5d7d210fbc4d8dc/jannovar-core/src/main/java/de/charite/compbio/jannovar/annotation/builders/DeletionAnnotationBuilder.java#L246

But the following lines might be also problematic! We have to evaluate that more closely

**Comment by holtgrewe** [comment]
@visze Is this resolved?

**Comment by visze** [comment]
No. I tried to fix it. But it seems to get really complicated!

Am Fr., 24. Aug. 2018 um 10:40 Uhr schrieb Manuel Holtgrewe <
notifications@github.com>:

**Comment by holtgrewe** [comment]
Fixed in 6f6f8b4c
